pymeasure/instruments/thorlabs/pm100usb.py

```
@@ -0,0 +1,97 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+from pymeasure.instruments import Instrument
+
+
+class ThorlabsPM100USB(Instrument):
+    """Represents Thorlabs PM100USB powermeter"""
+
+    # TODO: refactor to check if the sensor wavelength is adjustable
+    wavelength = Instrument.control("SENSE:CORR:WAV?", "SENSE:CORR:WAV %g",
+                                    "Wavelength in nm; not set outside of range")
+    
+    # TODO: refactor to check if the sensor is a power sensor
+    power = Instrument.measurement("MEAS:POW?", "Power, in W")
+
+    def __init__(self, adapter, **kwargs):
+        super(ThorlabsPM100USB, self).__init__(
+            adapter, "ThorlabsPM100USB powermeter", **kwargs)
+        self.timout = 3000
+        self.sensor()
+        self.wavelength_range()
+
+    def meas_power(self, wavelength):
+        """Set wavelength in nm and get power in W
+        If wavelength is out of range it will not be setted of range"""
+        if wavelength < self.wave_min:
+            print("Wavelength %f nm out of range: using minimum wavelength: %f nm" % (
```

Please raise an `Exception` instead of `print` when there is an error.